### PR TITLE
BandEdge orbitals and hybridization dos featurizer

### PR DESCRIPTION
**Pull request by albalu**
introducing features like cbm_s, vbm_sp (for hybridization), vbm_Si_d

these features are always present in BandEdge as opposed to dominant contributors only that is returned in DosFeaturizer.

### CODE REVIEW

File: matminer/featurizers/dos.py [hunk 1]
```diff
@@ -186,6 +192,95 @@ def citations(self):
...
+        for ex in ['cbm', 'vbm']:
+            for hybrid in ['sp', 'sd', 'sf', 'pd', 'pf', 'df']:
+                orb1 = feat['{}_{}'.format(ex, hybrid[0])]
+                orb2 = feat['{}_{}'.format(ex, hybrid[1])]
+                feat['{}_{}'.format(ex, hybrid)] = (orb1 * orb2) * 4.0  # 4x so max=1.0
+        return list(feat.values())
+
+    def feature_labels(self):
+        """
+        Returns ([str]): feature names starting with the extremun (cbm or vbm)
```
[the remark this answers is not in the document]
Comment (albalu) [reply matminer/featurizers/dos.py]: ok sure

**Comment by albalu** [comment]
class doc is missing, not ready to merge

**Comment by albalu** [comment]
ok this should be ready.

File: matminer/featurizers/dos.py [hunk 2]
```diff
@@ -186,13 +201,116 @@ def citations(self):
...
+
+    Returns (featurize returns [float] and featurize_labels returns [str]):
+        set of orbitals contributions and hybridizations and
+        possibly contributions from given species. Some examples for each
+        type of features:
+            cbm_s (float): s-orbital character of the cbm up to energy_cutoff
+            vbm_sp (float): sp-hybridization at the vbm edge. Minimum is 0
+                or no hybridization (e.g. all s or vbm_s==1) and 1.0 is
+                maximum hybridization (i.e. vbm_s==0.5, vbm_p==0.5)
+            cbm_Si_f (float): f-orbital character of Si
```
Comment (computron) [inline matminer/featurizers/dos.py]: Si doesn't have f orbitals, bad example ...

Comment (computron) [reply matminer/featurizers/dos.py]: I'll just change after merging

**Pull request merged**